HomeTask/HomeTask/Model/CoreDataManager.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  TasksPersistence.swift
+//  HomeTask
+//
+//  Created by Lucas Migge de Barros on 11/01/23.
+//
+
+import Foundation
+import CoreData
+
+
+protocol TaskPersistence {
+    func fetchTaks(completion: @escaping ([TaskItem]) -> Void)
+    
+    func saveTask(taskTitle: String, completion: @escaping () -> Void)
+    
+    func deleteTask(forTask task: TaskItem, completion: @escaping () -> Void)
+}
```

Comment:
Nada bloqueante, mas pensando em como podemos escrever esses métodos mais voltamos para as guidelines de design de API de Swift ([Swift API Design Guidelines](https://www.swift.org/documentation/api-design-guidelines/)), os dois últimos poderiam ser levemente ajustados para ficarem mais legíveis como de fato uma frase em inglês:
```suggestion
protocol TaskPersistence {
    func fetchTasks(completion: @escaping ([TaskItem]) -> Void)
    
    func saveTask(withTitle taskTitle: String, completion: @escaping () -> Void)
    
    func delete(task: TaskItem, completion: @escaping () -> Void)
}
```

Reply:
sim! faz sentido!